include/mcl_3dl/noise_generator.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2019, the mcl_3dl authors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef MCL_3DL_NOISE_GENERATOR_H
+#define MCL_3DL_NOISE_GENERATOR_H
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <random>
+#include <vector>
+
+#include <Eigen/Core>
+#include <Eigen/Eigenvalues>
+
+#include "ros/ros.h"
+
+namespace mcl_3dl
+{
+template <typename FLT_TYPE>
+class DiagonalNoiseGenerator
+{
+public:
+  template <typename T>
+  explicit DiagonalNoiseGenerator(T& sigma)
```

Comment:
I think it's better to handle mean value of the distribution in noise generator class.
It will make it easy to implement other random distribution types like uniform distribution in desired polygons which would be useful to generate initial distribution.

Reply:
Mean is used in State6DOF::generateNoise() (i.e. out of noise generator classes), so it is needed to update interfaces to do it. I made new issue (#260).

Reply:
API of `generateNoise` is changed in this PR and it will be changed again to do #260.
IMO, it would better be done in this PR to minimize the number of API change.
At least, it should be done before the next release.